### PR TITLE
Reduce Nexus example to define extension inline

### PR DESCRIPTION
**Pull request by dcousens**
This pull request reduces the complexity of the Nexus example to something that would be more canonically written as an `extendGraphqlSchema` function.

This format is inline with how we are using `extendGraphqlSchema` in [extend-graphql-schema-graphql-ts](https://github.com/keystonejs/keystone/blob/c6c9078d5cdeab968d563c340904717ebc36b1a3/examples/extend-graphql-schema-graphql-ts/schema.ts#L34) and [extend-graphql-schema-graphql-tools](https://github.com/keystonejs/keystone/blob/main/examples/extend-graphql-schema-graphql-tools/schema.ts#L35).

This is a followup to https://github.com/keystonejs/keystone/pull/6792

### CODE REVIEW

File: examples/extend-graphql-schema-nexus/nexus/types/Post.ts [hunk 1]
```diff
@@ -1,46 +0,0 @@
...
-
-export const NexusPost = objectType({
-  name: 'NexusPost',
-  definition(t) {
-    t.string('id');
-    t.string('title');
-    t.string('status');
-    t.string('content');
-  },
-});
```
Comment (dcousens) [inline examples/extend-graphql-schema-nexus/nexus/types/Post.ts:21]: This wasn't actually required, we needed to use https://github.com/graphql-nexus/nexus/pull/983

File: examples/extend-graphql-schema-nexus/schema.ts [hunk 2]
```diff
@@ -28,3 +32,63 @@ export const lists = {
...
+    mergeSchema: {
+      schema: baseSchema,
+    },
+    types: {
+      NexusThing,
+      NexusPostQuery,
+      NexusThingQuery,
+    },
+  });
+}
```
Comment (dcousens) [inline examples/extend-graphql-schema-nexus/schema.ts:94]: I think this example shows that `nexus` is ~equivalent to our `graphql-ts` example?

File: examples/extend-graphql-schema-nexus/schema.ts [hunk 3]
```diff
@@ -40,18 +40,20 @@ export function extendGraphqlSchema(baseSchema: GraphQLSchema) {
       t.field('nexusPosts', {
         type: nexus.nonNull(nexus.list('Post')),
         args: {
-          authorId: nexus.stringArg(),
+          id: nexus.nonNull(nexus.stringArg()),
```
Comment (dcousens) [inline examples/extend-graphql-schema-nexus/schema.ts:43]: This is inline with our other examples

File: examples/extend-graphql-schema-nexus/schema.ts [hunk 4]
```diff
@@ -85,6 +87,13 @@ export function extendGraphqlSchema(baseSchema: GraphQLSchema) {
     mergeSchema: {
       schema: baseSchema,
     },
+    outputs: {
+      typegen: path.join(process.cwd(), 'nexus-types.ts'),
+    },
+    contextType: {
+      module: path.join(process.cwd(), 'node_modules', '.keystone', 'types.d.ts'),
+      export: 'Context',
```
Comment (dcousens) [inline examples/extend-graphql-schema-nexus/schema.ts:95]: I don't know if I prefer this to `context: Context`, but, it's how nexus works

File: examples/extend-graphql-schema-nexus/schema.ts [hunk 5]
```diff
@@ -28,3 +32,75 @@ export const lists = {
...
+      schema: baseSchema,
+    },
+    types: {
+      NexusThing,
+      NexusPostQuery,
+      NexusThingQuery,
+    },
+
+    // Typescript output settings, probably something you might commit in dev
+    shouldGenerateArtifacts: process.env.NODE_ENV !== 'production',
```
Comment (dcousens) [inline examples/extend-graphql-schema-nexus/schema.ts:97]: Outputs with `yarn build` only (webpack)
`yarn dev` doesn't work, probably something to do with `esbuild`

**Review by borisno2 — APPROVED** [review]

**Pull request merged**